subprojects/core-api/src/main/java/org/gradle/api/provider/Provider.java

```diff
@@ -144,4 +144,12 @@
      */
     @Incubating
     Provider<T> orElse(Provider<? extends T> provider);
+
+    /**
+     * Returns a view of this {@link Provider} which can be safely read at configuration time.
+     *
+     * @since 6.5
+     */
+    @Incubating
+    Provider<T> forUseAtConfigurationTime();
```

Comment:
should we define what happens if this is not possible? for example , when the provider value is derived from a task output and is not available until the task has completed?

Reply:
added follow-up issue: https://github.com/gradle/instant-execution/issues/329